src/libcollections/btree/node.rs

```diff
@@ -47,34 +50,26 @@ pub enum SearchResult<NodeRef> {
 pub struct Node<K, V> {
     // To avoid the need for multiple allocations, we allocate a single buffer with enough space
     // for `capacity` keys, `capacity` values, and (in internal nodes) `capacity + 1` edges.
-    // Despite this, we store three separate pointers to the three "chunks" of the buffer because
-    // the performance drops significantly if the locations of the vals and edges need to be
-    // recalculated upon access.
     //
-    // These will never be null during normal usage of a `Node`. However, to avoid the need for a
-    // drop flag, `Node::drop` zeroes `keys`, signaling that the `Node` has already been cleaned
+    // This will never be null during normal usage of a `Node`. However, to avoid the need for a
+    // drop flag, `Node::drop` zeroes `data`, signaling that the `Node` has already been cleaned
     // up.
-    keys: Unique<K>,
-    vals: Unique<V>,
-
-    // In leaf nodes, this will be null, and no space will be allocated for edges.
-    edges: Unique<Node<K, V>>,
+    data: *mut u8,
```

Comment:
This could probably become

```rust
enum Data {
    Internal(Unique<([K; 2*B-1], [V; 2*B-1], [Node<K, V>; 2*B])>),
    Leaf(Unique<([K; 2*B-1], [V; 2*B-1])>),
}
```

and incorporate the `is_leaf` bool into the data.